packages/vite/src/node/plugins/resolve.ts

```diff
@@ -872,18 +872,27 @@ export async function tryOptimizedResolve(
   // We should be able to remove this in the future
   await depsOptimizer.scanProcessing
 
+  // resolvedSrc is what the importer wants
+  // we can return a optimizedDep only if optimizedDep.src === resolvedSrc
+  // https://github.com/vitejs/vite/pull/11290
+  let resolvedSrc: string | undefined
+  const basedir = importer ? path.dirname(importer) : process.cwd()
+  try {
+    // this may throw errors if unable to resolve, e.g. aliased id
+    resolvedSrc = normalizePath(resolveFrom(id, basedir))
+  } catch {
+    // this is best-effort only so swallow errors
+  }
+  if (!resolvedSrc) return
```

Comment:
I think if `importer` is undefined, `tryOptimizedResolve` should be allowed to match loosely for the doppelganger. Only if `importer` is defined, we skip the fast-path and go through the for loop instead.

Ideally to avoid calling `process.cwd()` when trying to resolve it.

So I think the change would be like before, but:

```js
  const metadata = depsOptimizer.metadata

  if (!importer) {
    const depInfo = optimizedDepInfoFromId(metadata, id)
    if (depInfo) {
      return depsOptimizer.getOptimizedDepId(depInfo)
    }
    return
  }
```

And we also get to skip `resolveFrom` if the `id` is never in the `depInfoList`

Reply:
Just added back the fast-path (loosely match without calling resolveFrom) for `import === undefined` case.